### PR TITLE
Add ssh url scheme handling

### LINKED ISSUE #2039: Support ssh:// url scheme handling
Are there any plans to support opening `ssh://` links.? 

I have tired setting Hyper as the handler with [duti](https://github.com/moretension/duti) which opens the app but does not ssh to the address. It would be nice if Hyper could set itself as the default handler like iTerm does.

### PR DESCRIPTION
**Pull request by cyrilico**
Fixes #2039 
PR is ready to be merged

Tested and working in macOS.
App also sets itself as the default ssh protocol client (only need to change defaultSSHClient boolean in app/config/config-default.js if user desires otherwise).
The URL components are retrieved using some simple regular expressions, it currently supports URLs of the type ssh://[username@]hostname[:port][[/]initial/remote/path]

Here is a short video showing the feature working: https://www.youtube.com/watch?v=gprDu_8n0KM

Note: Although it appears as if package.json was heavily modified, only a few small changes were made, however indentation issues made git recognise it as otherwise

### CODE REVIEW

**Comment by antonioalmeida** [comment]
Travis CI's check failed because it timed out after 20 minutes downloading a file at very slow speeds, the previous commit passed without problems

**Comment by diogotorres97** [comment]
Probably this error is related with [Travis CI Status](https://www.traviscistatus.com/incidents/95yttvs37v25)

**Comment by cyrilico** [comment]
@albinekb I believe all is ok now (once again)

**Comment by chabou** [comment]
I must test it against HyperCLI and test if we can add multiple URL schemes.
I will do it ASAP

File: app/index.js [hunk 1]
```diff
@@ -172,6 +172,11 @@ app.on('ready', () =>
       // expose to plugins
       app.createWindow = createWindow;
 
+      // check if should be set as default ssh protocol client
+      if (config.getConfig().defaultSSHApp == true && !app.isDefaultProtocolClient('ssh')) {
```
Comment (ppot) [inline app/index.js]: you can just do `config.getConfig().defaultSShApp` no need to test if `true` or else if you need to implicitly test it you will need to `config.getConfig().defaultSShApp === true`

Comment (antonioalmeida) [reply app/index.js]: Yup, my bad, just fixed it

**Comment by diogotorres97** [comment]
@chabou What is the current situation?

**Comment by diogotorres97** [comment]
@albinekb rebase done!

File: lib/actions/ui.js [hunk 2]
```diff
@@ -279,6 +280,45 @@ export function openFile(path) {
...
+    dispatch({
+      type: UI_OPEN_SSH_URL,
+      effect() {
+        let user = '',
+          port = '',
+          path = '';
+
+        let userRegExp = /ssh:\/\/((?:\w+@)?[\w.]+)(?=:.*)?/;
+        let portRegExp = /.*:(\d+)(?=\/.*)?/;
+        let pathRegExp = /ssh:\/\/.*:(?:\d+)?(\/?.*)/;
```
[the remark this answers is not in the document]
Comment (antonioalmeida) [reply lib/actions/ui.js]: Yes, the regexp doesn't work on that case. I'll look into that package, but it's probably an overkill for this usage alone.

File: lib/actions/ui.js [hunk 3]
```diff
@@ -279,6 +280,45 @@ export function openFile(path) {
...
+
+        regExec = portRegExp.exec(url);
+        if (regExec != null) port = regExec[1];
+
+        regExec = pathRegExp.exec(url);
+        if (regExec != null) path = regExec[1];
+
+        let command = 'ssh ' + user;
+        if (port != '') command += ' -p ' + port;
+        if (path != '') command += ' -t "' + path + '; bash"';
```
[the remark this answers is not in the document]
Comment (antonioalmeida) [reply lib/actions/ui.js]: Yeah, this is an ugly workaround, better to ignore paths.

**Comment by antonioalmeida** [comment]
Update:
- ssh url parsing is now made using [parse-url](https://www.npmjs.com/package/parse-url)
- path is now ignored
@chabou

**Comment by chabou** [comment]
@antonioalmeida I added 2 commits:
* First one to remove Hyper from default client for `ssh://` protocol when user change its config.
* Second one to deal with parse-url dependency. We only use `yarn` (with pinned dependency) and its `yarn.lock` lockfiles. And we need to add package to `app/package.json` when it is used by main process.

It works great. Thank you so much 🙏 . I'll merge it ASAP.

**Pull request merged**